WooCommerce/Classes/ViewModels/ProductDetailsCellViewModel.swift

```diff
@@ -56,54 +23,73 @@ struct ProductDetailsCellViewModel {
 
     /// Item Quantity as a String
     ///
-    var quantity: String {
-        return NumberFormatter.localizedString(from: positiveQuantity as NSDecimalNumber, number: .decimal)
-    }
+    let quantity: String
 
     /// The localized total value of this line item. This is the quantity x price.
     ///
-    var total: String {
-        currencyFormatter.formatAmount(positiveTotal, with: currency) ?? String()
-    }
+    let total: String
 
     /// Returns the localized "quantity x price" to use for the subtitle.
     ///
-    var subtitle: String {
-        let itemPrice = currencyFormatter.formatAmount(positivePrice, with: currency) ?? String()
-
-        return Localization.subtitle(quantity: quantity, price: itemPrice, attributes: attributes)
-    }
+    let subtitle: String
 
     /// Item SKU
     ///
-    var sku: String? {
-        guard let sku = skuText, sku.isEmpty == false else {
-            return nil
-        }
+    let sku: String?
 
-        let skuTemplate = NSLocalizedString("SKU: %@", comment: "SKU label, followed by the SKU")
-        let skuText = String.localizedStringWithFormat(skuTemplate, sku)
+    // MARK: - Initializers
 
-        return skuText
-    }
+    private init(currency: String,
+                 currencyFormatter: CurrencyFormatter,
+                 imageURL: URL?,
+                 name: String,
+                 positiveQuantity: Decimal,
+                 positiveTotal: NSDecimalNumber?,
+                 positivePrice: NSDecimalNumber?,
+                 skuText: String?,
+                 attributes: [OrderAttributeViewModel]) {
+        self.imageURL = imageURL
+        self.name = name
+        let quantity = NumberFormatter.localizedString(from: positiveQuantity as NSDecimalNumber, number: .decimal)
+        self.quantity = quantity
+        self.sku = {
+            guard let sku = skuText, sku.isEmpty == false else {
+                return nil
+            }
+            return String.localizedStringWithFormat(Localization.skuFormat, sku)
+        }()
 
-    // MARK: - Initializers
+        self.total = {
+            guard let positiveTotal = positiveTotal else {
+                return ""
+            }
+            return currencyFormatter.formatAmount(positiveTotal, with: currency) ?? String()
```

Comment:
Great question! `String()` and `""` are the same, according to the [Apple doc](https://docs.swift.org/swift-book/LanguageGuide/StringsAndCharacters.html) under "Initializing an Empty String":

> var emptyString = ""               // empty string literal
var anotherEmptyString = String()  // initializer syntax
// these two strings are both empty, and are equivalent to each other

I'll update the empty string literal to `String()` to avoid confusion 😄

Reply:
Updated in: https://github.com/woocommerce/woocommerce-ios/pull/3254/commits/ad204b4390d4ae2165acb43535e4fb2f5865bc8a